src/main/java/com/google/api/codegen/transformer/csharp/CSharpApiMethodTransformer.java

```diff
@@ -86,4 +110,118 @@ protected void setServiceResponseTypeName(
             "Cannot handle streaming type: " + context.getMethodConfig().getGrpcStreamingType());
     }
   }
+
+  @Override
+  public List<StaticLangApiMethodView> generateApiMethods(InterfaceContext interfaceContext) {
+    Preconditions.checkArgument(
+        interfaceContext instanceof GapicInterfaceContext,
+        "SampleGen is for protobuf-based API" + " only.");
+    GapicInterfaceContext context = (GapicInterfaceContext) interfaceContext;
+    List<ParamWithSimpleDoc> pagedMethodAdditionalParams =
+        new ImmutableList.Builder<ParamWithSimpleDoc>()
+            .addAll(csharpCommonTransformer.pagedMethodAdditionalParams())
+            .addAll(csharpCommonTransformer.callSettingsParam())
+            .build();
+
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+    for (MethodModel method : csharpCommonTransformer.getSupportedMethods(context)) {
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      MethodContext requestMethodContext = context.asRequestMethodContext(method);
+      if (methodConfig.isGrpcStreaming()) {
+        // Only for protobuf-based APIs.
```

Comment:
Why do we need this comment? (I realize this is moved from the `CSharpGapicClientTransformer`, but I'm still curious)

Reply:
I  _think_ that means it does not apply to discogapics.
That said, looking a few lines above, we have already checked `interfaceContext` is a `GapicInterfaceContext`. So I do think this comment is redundant. I'll remove it in a coming commit.
(Also noticed the error message of that `checkArgument` is not appropriate, will change that too)